src/openai/openai.service.ts

```diff
@@ -0,0 +1,88 @@
+import { Injectable, Logger  } from "@nestjs/common";
+import OpenAIApi  from "openai";
+import { apiRequestDTO, Message } from "./dto/apiRequest.dto";
+import { GuideResponseDTO} from "./dto/guideResponse.dto";
+import {
+    GPT_MODEL, 
+    SYSTEM_MESSAGE, 
+    USER_PROMPT, 
+    OUTPUT_INDICATOR, 
+    CONTEXT_EXAMPLE, 
+} from "./constants/constants";
+import { ConfigService } from "@nestjs/config";
+
+@Injectable()
+export class OpenAIService {
+    private readonly openai: OpenAIApi;
+    private readonly configService: ConfigService;
+    
+    constructor() {
+        this.configService = new ConfigService();
+        this.openai = new OpenAIApi({ apiKey: this.configService.get('OPENAI_API_KEY') });
+    }
+
+
+    formatConversation (requestDto: apiRequestDTO): Message[] {
+        console.log("formatConversation: requestDto: ", requestDto);
+        const chatId = requestDto.chatId;
+        const customerId = requestDto.user.userId;
+        const userName = requestDto.user.username;
```

Comment:
I dont think you need all this data from the frontend, you only need the messages array in order to create a guide.

Reply:
i need some User data like userName and role.